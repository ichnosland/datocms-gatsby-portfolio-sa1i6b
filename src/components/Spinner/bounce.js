import styled, { css, keyframes } from 'styled-components';

import { colore } from 'style/color';
import media from 'style/mediainjector';

export const delay = (turn = 0.0) => ` -${turn * 0.16}s`;

const bounce = keyframes`
  0%, 80%, 100% {
    -webkit-transform: scale(0);
    transform: scale(0);
  } 40% {
    -webkit-transform: scale(1.0);
    transform: scale(1.0);
  }
`;

export const dimension = (size) => {
  switch (size) {
    case 'small':
      return '10px';
    case 'huge':
      return '30px';
    default:
      return '18px';
  }
};

export const Bounce = styled.div`
  width: ${(props) => dimension(props.size)};
  height: ${(props) => dimension(props.size)};
  background-color: ${(props) => props.bgColor ? props.bgColor : props.theme.brand};
  ${({ inverse }) => inverse && css`background-color: ${colore.ui.contrast};`}
  ${media.mobile`
    ${(props) => props.ringo && css`
      background-color: ${colore.ui.contrast};
    `};
  `};
  border-radius: 100%;
  display: inline-block;
  animation: ${bounce} 1.4s infinite ease-in-out both;
  animation-delay: ${(props) => props.number ? delay(props.number) : 0};
`;

Bounce.defaultProps = {
  theme: {
    brand: '#00BBEF',
  },
};
