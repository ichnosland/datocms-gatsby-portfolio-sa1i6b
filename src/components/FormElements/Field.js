/**
*
* Field
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'styled-components';
import { colore } from 'style/color';
import FlexBox from '../FlexBox';
import Spinner from '../Spinner';
import { BasicInputStyle, FormMessage } from './index';

export const Input = styled.input`
  ${BasicInputStyle};
  &[type="search"] {
    box-sizing: border-box;
    text-align: left;
    border-radius: ${(props) => props.theme.radius.searchBox};
    background-color: ${colore.ui.contrast};
  }
  ${(props) => props.landing && css`
    order: 2;
    margin-bottom: 12px;
    width: 100%;
    appearance: none;
    background-color: white;
    background-image: none;
    padding: 6px 13px 6px 13px;
    border: none;
    border-radius: 3px;
    &::placeholder {
      position: relative;
      color: rgba(0, 0, 0, 0.3);
      opacity:1;
      overflow: visible;
    }
  `}
`;

Input.defaultProps = {
  theme: {
    brand: '#00BBEF',
    subtle: '#B4E1FF',
    light: '#C9E4FF',
    radius: {
      searchBox: '100px',
    },
  },
};

class Field extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { input, note, align, label, type, meta: { asyncValidating, touched, error }, landing } = this.props;
    return (
      <FlexBox direction="column">
        <Input
          {...input}
          placeholder={label}
          type={type}
          note={note}
          align={align}
          landing={landing}
        />
        {asyncValidating && <Spinner size="small" />}
        {touched && error && <FormMessage landing={landing}>{error}</FormMessage>}
      </FlexBox>
    );
  }
}

Input.defaultProps = {
  theme: {
    radius: {
      searchBox: '100px',
    },
  },
};

Field.propTypes = {
  input: PropTypes.object.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  meta: PropTypes.object.isRequired,
  note: PropTypes.bool,
  landing: PropTypes.bool,
  align: PropTypes.string,
};

export default Field;
