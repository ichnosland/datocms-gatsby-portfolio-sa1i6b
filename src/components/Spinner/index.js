/**
*
* Spinner
*
*/

import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Div from 'components/Div';
import { Bounce } from './bounce';

const SpinnerWrap = styled(Div)`
  margin-top: 100px;
  text-align: center;
`;

class Spinner extends React.PureComponent { // eslint-disable-line react/prefer-stateless-function
  render() {
    const { number, bgColor, size, ringo } = this.props;
    const bounces = [...Array(number).keys()].reverse();

    return (
      <SpinnerWrap>
        {bounces.map(
          (n) => <Bounce key={n} number={n} bgColor={bgColor} size={size} ringo={ringo} />
        )}
      </SpinnerWrap>
    );
  }
}

Spinner.propTypes = {
  number: PropTypes.number,
  bgColor: PropTypes.string,
  size: PropTypes.string,
  ringo: PropTypes.bool,
};

Spinner.defaultProps = {
  number: 3,
};

export default Spinner;
