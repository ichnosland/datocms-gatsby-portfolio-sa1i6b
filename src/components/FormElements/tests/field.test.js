import React from 'react';
import { shallow } from 'enzyme';
import Spinner from '../Spinner';
import Field, { Input } from '../Field';
import { FormMessage } from '../index';

const props = {
  input: {},
  label: 'placeholder',
  type: 'text',
  meta: {},
};
const theme = {
  brand: '#111111',
  light: '#333333',
  subtle: '#ffffff',
  radius: {
    searchBox: '80px',
    general: '6px',
  },
};

describe('<Field /> ', () => {
  it('should render an Input', () => {
    const renderedComponent = shallow(<Field {...props} />);
    expect(renderedComponent.find(Input).length).toEqual(1);
  });

  it('should render the placeholder', () => {
    const renderedComponent = shallow(<Field {...props} />);
    expect(renderedComponent.find(Input).prop('placeholder')).toBe(props.label);
  });

  it('should render the error', () => {
    const rendered = {
      ...props,
      meta: {
        touched: true,
        error: 'un errore',
      },
    };
    const error = <FormMessage>{rendered.meta.error}</FormMessage>;
    const renderedComponent = shallow(<Field {...rendered} />);
    expect(renderedComponent.contains(error)).toBe(true);
  });

  it('should render the Spinner', () => {
    const rendered = {
      ...props,
      meta: {
        asyncValidating: true,
      },
    };
    const renderedComponent = shallow(<Field {...rendered} />);
    expect(renderedComponent.contains(<Spinner size="small" />)).toBe(true);
  });

  it('should render its css w/ props note', () => {
    const renderedComponent = shallow(<Field {...props} note />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props align', () => {
    const align = 'right';
    const renderedComponent = shallow(<Field {...props} align={align} />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

describe('<Input />', () => {
  it('should render a <input> tag', () => {
    const renderedComponent = shallow(<Input />);
    expect(renderedComponent.type()).toEqual('input');
  });

  it('should render its css w/ props theme', () => {
    const renderedComponent = shallow(<Input theme={theme} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props note', () => {
    const renderedComponent = shallow(<Input note />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props align', () => {
    const align = 'center';
    const renderedComponent = shallow(<Input align={align} />);
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should render its css w/ props landing', () => {
    const renderedComponent = shallow(<Input landing />);
    expect(renderedComponent).toMatchSnapshot();
  });
});

