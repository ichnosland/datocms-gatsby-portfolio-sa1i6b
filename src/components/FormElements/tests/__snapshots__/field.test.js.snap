// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field />  should render its css w/ props align 1`] = `
<FlexBox
  direction="column"
>
  <Field__Input
    align="right"
    placeholder="placeholder"
    theme={
      Object {
        "radius": Object {
          "searchBox": "100px",
        },
      }
    }
    type="text"
  />
</FlexBox>
`;

exports[`<Field />  should render its css w/ props note 1`] = `
<FlexBox
  direction="column"
>
  <Field__Input
    note={true}
    placeholder="placeholder"
    theme={
      Object {
        "radius": Object {
          "searchBox": "100px",
        },
      }
    }
    type="text"
  />
</FlexBox>
`;

exports[`<Input /> should render its css w/ props align 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  width: 100%;
  text-align: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 6px 13px;
  border: 1px solid;
  background-color: rgb(255,255,255);
}

.c0::-webkit-input-placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0::-moz-placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0:-ms-input-placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0::placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0:active,
.c0:focus {
  outline: none;
}

.c0[type="search"] {
  box-sizing: border-box;
  text-align: left;
  border-radius: 100px;
  background-color: rgb(255,255,255);
}

<input
  className="c0"
/>
`;

exports[`<Input /> should render its css w/ props landing 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  width: 100%;
  text-align: left;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 6px 13px;
  border: 1px solid;
  background-color: rgb(255,255,255);
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  margin-bottom: 12px;
  width: 100%;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background-color: white;
  background-image: none;
  padding: 6px 13px 6px 13px;
  border: none;
  border-radius: 3px;
}

.c0::-webkit-input-placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0::-moz-placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0:-ms-input-placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0::placeholder {
  position: relative;
  opacity: 1;
  overflow: visible;
}

.c0:active,
.c0:focus {
  outline: none;
}

.c0[type="search"] {
  box-sizing: border-box;
  text-align: left;
  border-radius: 100px;
  background-color: rgb(255,255,255);
}

.c0::-webkit-input-placeholder {
  position: relative;
  color: rgba(0,0,0,0.3);
  opacity: 1;
  overflow: visible;
}

.c0::-moz-placeholder {
  position: relative;
  color: rgba(0,0,0,0.3);
  opacity: 1;
  overflow: visible;
}

.c0:-ms-input-placeholder {
  position: relative;
  color: rgba(0,0,0,0.3);
  opacity: 1;
  overflow: visible;
}

.c0::placeholder {
  position: relative;
  color: rgba(0,0,0,0.3);
  opacity: 1;
  overflow: visible;
}

<input
  className="c0"
/>
`;

exports[`<Input /> should render its css w/ props note 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  width: 100%;
  text-align: left;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 6px 13px;
  border: 1px solid;
  background-color: rgb(255,255,255);
  color: rgb(78,78,78);
  background-color: #FFFDDE;
  border-color: #D5D3C1;
}

.c0::-webkit-input-placeholder {
  position: relative;
  color: #D5D3C1;
  opacity: 1;
  overflow: visible;
}

.c0::-moz-placeholder {
  position: relative;
  color: #D5D3C1;
  opacity: 1;
  overflow: visible;
}

.c0:-ms-input-placeholder {
  position: relative;
  color: #D5D3C1;
  opacity: 1;
  overflow: visible;
}

.c0::placeholder {
  position: relative;
  color: #D5D3C1;
  opacity: 1;
  overflow: visible;
}

.c0:active,
.c0:focus {
  outline: none;
}

.c0[type="search"] {
  box-sizing: border-box;
  text-align: left;
  border-radius: 100px;
  background-color: rgb(255,255,255);
}

<input
  className="c0"
/>
`;

exports[`<Input /> should render its css w/ props theme 1`] = `
.c0 {
  display: block;
  box-sizing: border-box;
  color: #111111;
  width: 100%;
  text-align: left;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 6px 13px;
  border: 1px solid #333333;
  background-color: rgb(255,255,255);
}

.c0::-webkit-input-placeholder {
  position: relative;
  color: #ffffff;
  opacity: 1;
  overflow: visible;
}

.c0::-moz-placeholder {
  position: relative;
  color: #ffffff;
  opacity: 1;
  overflow: visible;
}

.c0:-ms-input-placeholder {
  position: relative;
  color: #ffffff;
  opacity: 1;
  overflow: visible;
}

.c0::placeholder {
  position: relative;
  color: #ffffff;
  opacity: 1;
  overflow: visible;
}

.c0:active,
.c0:focus {
  outline: none;
}

.c0[type="search"] {
  box-sizing: border-box;
  text-align: left;
  border-radius: 80px;
  background-color: rgb(255,255,255);
}

<input
  className="c0"
/>
`;
